### PR TITLE
pagination when selecting files in desktop app [sc-1471]

### PR DESCRIPTION
**Pull request by operramon**

### CODE REVIEW

File: apps/desktop/src/app/upload/select-files/select-files.component.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    ),
+    share()
+  );
+
+  constructor(private cdr: ChangeDetectorRef) { }
+
+  ngAfterViewInit() {
+    setTimeout(() => { this.triggerSearch.next() }, 200);
+
+    // TODO: can't use async pipe in the template because (unexpectedly) it doesn't work on GDrive connector.
```
Comment (ebrehault) [inline apps/desktop/src/app/upload/select-files/select-files.component.ts:63]: I had some weird things with GDrive too, not sure exactly what it is about, but it is not critical anyway.

File: apps/desktop/src/app/upload/select-files/select-files.component.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        this.items = resources;
+        this.cdr.detectChanges();
+      });
+
+    // Infinite scroll
+    const container = this.scroll?.nativeElement;
+    if (container) {
+      merge(fromEvent(container, 'scroll'), this.resources)
+        .pipe(
+          auditTime(300),
```
Comment (ebrehault) [inline apps/desktop/src/app/upload/select-files/select-files.component.ts:76]: Cool stuff! I've never heard about it before.

**Review by ebrehault — APPROVED** [review]

File: apps/desktop/src/app/sync/sources/dropbox.ts [hunk 3]
```diff
@@ -149,6 +163,16 @@ class DropboxImpl implements ISourceConnector {
     };
   }
 
+  /* eslint-disable  @typescript-eslint/no-explicit-any */
+  private filterFiles(raw: any): boolean {
```
[the remark this answers is not in the document]
Comment (operramon) [reply apps/desktop/src/app/sync/sources/dropbox.ts:167]: thanks! I didn't know that.
when changing `any` to `unknown` another error appears:
```
Object is of type 'unknown'
```
I guess typescript rules are very strict

[the remark this answers is not in the document]
Comment (mpellerin42) [reply apps/desktop/src/app/sync/sources/dropbox.ts:167]: oh, maybe it depends on the version of typescript and/or on the rules configured on eslint? 🤔
I remember I used that trick a lot in my previous job, I'll try to dig a bit more to find out if there is one better than the other or not. In any case, this isn't a blocker for your PR at all.

**Pull request merged**